src/Altinn.App.Core/Infrastructure/Clients/Storage/ProcessLockClient.cs

```diff
@@ -0,0 +1,99 @@
+        IOptions<PlatformSettings> platformSettings,
+        IOptions<AppSettings> appSettings,
+        IHttpContextAccessor httpContextAccessor,
+        HttpClient httpClient,
+        Telemetry? telemetry = null
+    )
+    {
+        _appSettings = appSettings.Value;
+        _httpContextAccessor = httpContextAccessor;
+        httpClient.BaseAddress = new Uri(platformSettings.Value.ApiStorageEndpoint);
+        httpClient.DefaultRequestHeaders.Add(General.SubscriptionKeyHeaderName, platformSettings.Value.SubscriptionKey);
+        httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+        _client = httpClient;
+        _telemetry = telemetry;
+    }
+
+    public async Task<Guid> AcquireProcessLock(Guid instanceGuid, int instanceOwnerPartyId, TimeSpan expiration)
+    {
+        using var activity = _telemetry?.StartAcquireProcessLockActivity(instanceGuid, instanceOwnerPartyId);
+        string apiUrl = $"instances/{instanceOwnerPartyId}/{instanceGuid}/process/lock";
+        string token = JwtTokenUtil.GetTokenFromContext(
+            _httpContextAccessor.HttpContext,
+            _appSettings.RuntimeCookieName
+        );
+
+        var request = new ProcessLockRequest { Expiration = (int)expiration.TotalSeconds };
+        var content = JsonContent.Create(request);
+
+        HttpResponseMessage response = await _client.PostAsync(token, apiUrl, content);
+
+        if (!response.IsSuccessStatusCode)
+        {
+            throw await PlatformHttpException.CreateAsync(response);
+        }
+
+        Guid? lockId = null;
+        try
+        {
+            var lockResponse = await response.Content.ReadFromJsonAsync<ProcessLockResponse>();
+            lockId = lockResponse?.LockId;
+        }
+        catch
+        {
+            // Throw exception below
+        }
+
+        if (lockId is null || lockId.Value == Guid.Empty)
+        {
+            throw new PlatformHttpException(
+                response,
+                "The response from the lock acquisition endpoint was not expected."
+            );
+        }
+
+        return lockId.Value;
+    }
+
+    public async Task ReleaseProcessLock(Guid instanceGuid, int instanceOwnerPartyId, Guid lockId)
+    {
+        using var activity = _telemetry?.StartReleaseProcessLockActivity(instanceGuid, instanceOwnerPartyId);
+        string apiUrl = $"instances/{instanceOwnerPartyId}/{instanceGuid}/process/lock/{lockId}";
+        string token = JwtTokenUtil.GetTokenFromContext(
+            _httpContextAccessor.HttpContext,
+            _appSettings.RuntimeCookieName
+        );
+
+        var request = new ProcessLockRequest { Expiration = 0 };
+        var content = JsonContent.Create(request);
+
+        HttpResponseMessage response = await _client.PatchAsync(token, apiUrl, content);
+
+        if (!response.IsSuccessStatusCode)
+        {
+            throw await PlatformHttpException.CreateAsync(response);
```

Comment:
`PlatformHttpResponseSnapshotException` was recently introduced, if we use that we can also dipose the response above